res_config_settings_enterprise_remove/models/res_config_settings.py

```diff
@@ -1,14 +1,58 @@
-# -*- coding: utf-8 -*-
 # Copyright 2016 LasLabs Inc.
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+# Copyright 2018 Onestein (<http://www.onestein.eu>)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
 
-from odoo import api, models
 from lxml import etree
 
+from odoo import api, models
+
+ENTERPRISE_MODULE_FIELD_NAMES = [
+    'account_hide_setup_bar',
+    'module_currency_rate_live',
+    'module_account_reports_followup',
+    'module_account_batch_deposit',
+    'module_account_sepa_direct_debit',
+    'module_l10n_us_check_printing',
+    'module_account_sepa',
+    'module_account_reports',
+    'module_account_yodlee',
+    'module_account_bank_statement_import_csv',
+    'module_account_bank_statement_import_qif',
+    'module_account_bank_statement_import_ofx',
+    'module_account_bank_statement_import_camt',
+    'module_print_docsaway',
+    'module_account_deferred_revenue',
+    'module_sale_coupon',
+    'module_delivery_ups',
+    'module_delivery_dhl',
+    'module_delivery_fedex',
+    'module_delivery_usps',
+    'module_delivery_bpost',
+    'module_stock_barcode',
+    'module_project_forecast',
+    'group_mrp_routings',
+    'module_pos_loyalty',
+    'module_web_clearbit',
+    'module_website_version',
+    'module_l10n_mx_edi',
+    'module_event_barcode',
+    'module_project_timesheet_synchro',
+    'module_account_accountant',
+    'module_account_3way_match',
+    'module_inter_company_rules',
+    'module_mrp_maintenance',
+    'module_quality_mrp',
+]
+
 
 class ResConfigSettings(models.TransientModel):
     _inherit = 'res.config.settings'
 
+    @classmethod
+    def _format_query_by_field_name(self, field_name):
+        return "//div[div[field[@name='%s' and \
+                @widget='upgrade_boolean']]]" % field_name
```

Comment:
Why not just search for the `@widget='upgrade_boolean'` part in loop until no more matches are found? 🤔 

That way we avoid maintaining the list of fields above...

Reply:
Thank you @Yajo ! Let's see how it behaves with that code simplification. I made a test and it seems working fine.